pyproject.toml

```diff
@@ -75,6 +75,7 @@ boto3 = { version = ">=1.24.59", optional = true }
 s3fs = { version = ">=2023.1.0", optional = true }
 adlfs = { version = ">=2023.1.0", optional = true }
 gcsfs = { version = ">=2023.1.0", optional = true }
+google-api-core = { version = "^2.0", optional = true }
```

Comment:
When do we need this one? Should it be part of one of the [`extras`](https://github.com/apache/iceberg-python/blob/4ccc9e39efa000e9023e3125f0cdf504e8755f75/pyproject.toml#L297)?